pkg/image/admission/imagepolicy/api/v1/types.go

```diff
@@ -0,0 +1,138 @@
+package v1
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/api/v1"
+)
+
+// ImagePolicyConfig is the configuration for control of images running on the platform.
+type ImagePolicyConfig struct {
+	unversioned.TypeMeta `json:",inline"`
+
+	// ExecutionRules determine whether the use of an image is allowed in an object with a pod spec.
+	// By default, these rules only apply to pods, but may be extended to other resource types.
+	// If all execution rules are negations, the default behavior is allow all. If any execution rule
+	// is an allow, the default behavior is to reject all.
+	ExecutionRules []ImageExecutionPolicyRule `json:"executionRules"`
+
+	// TODO: enable additional rule types
+	// ConsumptionRules are applied when creating resources with pod specs.
+	// ConsumptionRules []ImageConsumptionPolicyRule `json:"consumptionRules"`
+	// PlacementRules are applied when creating resources with pod specs.
+	// PlacementRules []ImagePlacementPolicyRule `json:"placementRules"`
+}
+
+// ImageExecutionPolicyRule determines whether a provided image may be used on the platform.
+type ImageExecutionPolicyRule struct {
+	ImageCondition `json:",inline"`
+
+	// Resolve indicates that images referenced by this resource must be resolved
+	Resolve bool `json:"resolve"`
+}
+
+// ImageCondition defines the conditions for matching a particular image source. The conditions below
+// are all required (logical AND). If Reject is specified, the condition is false if all conditions match,
+// and true otherwise.
+type ImageCondition struct {
+	// Name is the name of this policy rule for reference. It must be unique across all rules.
+	Name string `json:"name"`
+	// IgnoreNamespaceOverride prevents this condition from being overriden when the
+	// `alpha.image.policy.openshift.io/ignore-rules` is set on a namespace and contains this rule name.
+	IgnoreNamespaceOverride bool `json:"ignoreNamespaceOverride"`
+
+	// Reject indicates this rule is inverted - if all conditions below match, then this rule is considered to
+	// not match.
+	Reject bool `json:"reject"`
```

Comment:
I think you'd want to come up with the example of where you have two exclusions that combine predicates.  If I want to ban a list of registries, I can do so today without the complex joining logic (matching registries is an OR).  I don't want to go to full nested predicates - we can target specific use cases if necessary with `MatchAnyImageLabels` in the future, but full nested predicates and general boolean logic in an API is not yet required by the use case.